tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -144,9 +141,19 @@ def test_fail_on_multiple_drivers():
         extract_db_type_from_uri("mysql+pymsql+pyodbc://...")
 
 
-@pytest.fixture
-def store(tmp_path: Path) -> SqlAlchemyStore:
-    store = _get_store(tmp_path)
+@pytest.fixture(params=["normal", "migrate"])
+def store(tmp_path: Path, request):
+    db_uri = MLFLOW_TRACKING_URI.get() or f"{DB_URI}{tmp_path / 'temp.db'}"
+    artifact_uri = tmp_path / "artifacts"
+    artifact_uri.mkdir(exist_ok=True)
+    if request.param == "migrate":
+        # Test case where user has an existing DB with schema generated before MLflow 1.0,
+        # then migrate their DB
+        engine = sqlalchemy.create_engine(db_uri)
+        InitialBase.metadata.create_all(engine)
+        engine.dispose()
+        invoke_cli_runner(mlflow.db.commands, ["upgrade", db_uri])
```

Comment:
I'm not super familiar with what this setup intended originally (and I know this PR doesn't change the logic itself, just to move it to parametrized fixture), but it looks a bit weird test set up:

1. When `MLFLOW_TRACKING_URI` is set, this update the actual tracking database rather than creating a temporary db, is it safe? Also this migration command runs for every test case but only the first call takes effect.
2. When tracking URI is not set, the new DB is created under `tmp_path`, then the schema should always be up-to-date. So it just runs the same tests twice.

Apparently the `TestSqlAlchemyStoreMigrateDB` test suite was added years ago. I don't believe any of active developers have an database with old schema before MLflow 1.0..? Moreover this setup aggressively update the database schema pointed by MLFLOW_TRACKING_URI, so people who executed this test once should have the latest schema.

cc: @BenWilson2 @harupy if you have more detailed context:)

Reply:
Hi, Yuki. Thank you for reviewing it. I migrated the tests with a policy that retains the test functionalities as it is. But if it is true that active developers don't use this old version, I agree to remove this test-after migration case since it doubles the number of test cases.

Reply:
Let's remove the fixture condition for the ancient version of the DB. This is extreme legacy support that we don't need to handle anymore. Great find on this @B-Step62 !!

Reply:
I've removed the test cases for migration. Please take another look when you have time.